src/component/feature/StrategyTypes/GeneralStrategy/GeneralStrategy.tsx

```diff
@@ -1,190 +1,47 @@
-                                )}
-                            </div>
-                        );
-                    } else if (type === 'boolean') {
-                        const value = parseParameterString(parameters[name]);
-                        return (
-                            <div key={name}>
-                                <Tooltip
-                                    title={description}
-                                    placement="right-end"
-                                    arrow
-                                >
-                                    <FormControlLabel
-                                        label={name}
-                                        control={
-                                            <Switch
-                                                name={name}
-                                                onChange={handleSwitchChange.bind(
-                                                    this,
-                                                    name,
-                                                    value
-                                                )}
-                                                checked={value === 'true'}
-                                            />
-                                        }
-                                    />
-                                </Tooltip>
-                            </div>
-                        );
-                    } else {
-                        const value = parseParameterString(parameters[name]);
-                        return (
-                            <div key={name}>
-                                <TextField
-                                    rows={1}
-                                    placeholder=""
-                                    variant="outlined"
-                                    size="small"
-                                    style={{ width: '100%' }}
-                                    required={required}
-                                    disabled={!editable}
-                                    name={name}
-                                    label={name}
-                                    onChange={onChangeTextField.bind(
-                                        this,
-                                        name
-                                    )}
-                                    value={value}
-                                />
-                                {description && (
-                                    <p className={styles.helpText}>
-                                        {description}
-                                    </p>
-                                )}
-                            </div>
-                        );
-                    }
-                }
-            )}
-        </div>
+        <StyledContainer>
+            {strategyDefinition.parameters.map((definition, index) => (
+                <div key={index}>
```

Comment:
is the objectId function on definition a better key?

Reply:
I think `index` should be enough since the order of the parameters won't change.